### PR TITLE
docs: Add test specifications for `TokenUpdateTransaction`

### LINKED ISSUE #178: Document E2E tests: `TokenUpdateTransaction`
Document the E2E `TokenUpdateTransaction` tests in `test-specifications/`.

### PR DESCRIPTION
**Pull request by rwalworth**
**Description**:
This PR adds the tests for the `TokenUpdateTransaction` fields.

**NOTE**: This PR contains work contained in #228, so that PR should be reviewed and merged first.

**Related issue(s)**:

Fixes #178 

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: test-specifications/token-service/tokenUpdateTransaction.md [hunk 1]
```diff
@@ -0,0 +1,810 @@
...
+| 9       | Updates a mutable token with a valid key as its metadata key but doesn't sign with it                                       | tokenId=<VALID_TOKEN_ID_WITHOUT_METADATA_KEY>, metadataKey=<VALID_KEY>, commonTransactionParams.signers=[<ADMIN_KEY_OF_VALID_TOKEN_WITHOUT_METADATA_KEY>] | The token update fails with an TOKEN_HAS_NO_METADATA_KEY response code from the network.         | N                 |
+| 10      | Updates a mutable token with an invalid key as its metadata key                                                             | tokenId=<CREATED_MUTABLE_TOKEN_ID>, metadataKey=<INVALID_KEY>, commonTransactionParams.signers=[<CREATED_MUTABLE_TOKEN_ADMIN_KEY>]                        | The token update fails with an SDK internal error.                                               | N                 |
```
Comment (jsync-swirlds) [inline test-specifications/token-service/tokenUpdateTransaction.md:779]: We may want to add a test case to update the token with an empty `KeyList` as the metadata key.  That's how a "role" key can be removed in a token update.

We might want to include that in each of the "role" keys for update, to cover that "clear a key" behavior.

**Review by jsync-swirlds — COMMENTED** [review]
A few suggestions that might help cover the full specification.

File: test-specifications/token-service/tokenCreateTransaction.md [hunk 2]
```diff
@@ -0,0 +1,1158 @@
...
+| 13      | Creates a token with a fractional fee with a numerator of 9,223,372,036,854,775,806 (int64 max - 1)                  | name="testname", symbol="testsymbol", treasuryAccountId=<OPERATOR_ACCOUNT_ID>, customFees=[{feeCollectorAccountId=<OPERATOR_ACCOUNT_ID>, feeCollectorsExempt=false, fractionalFee.numerator=9223372036854775806, fractionalFee.denominator=10, fractionalFee.minimumAmount=1, fractionalFee.maximumAmount=10, fractionalFee.assessmentMethod="inclusive"}]                                                                                                                                 | The token creation succeeds and the token has the custom fractional fee with an amount of 9,223,372,036,854,775,806 / 10.             | N                 |
```
[the remark this answers is not in the document]
Comment (rwalworth) [reply test-specifications/token-service/tokenCreateTransaction.md:918]: The tests here mainly confirm that the platform is able to correctly parse these different overflow situations. This case will probably be covered in the `TransferTransaction` tests, when those get written.

[the remark this answers is not in the document]
Comment (jsync-swirlds) [reply test-specifications/token-service/tokenCreateTransaction.md:918]: Sure, just wondering if we want to describe the behavior here or just the test.

File: test-specifications/token-service/tokenCreateTransaction.md [hunk 3]
```diff
@@ -0,0 +1,1158 @@
...
+| 3       | Creates a token with a valid ED25519 private key as its admin key                                                | name="testname", symbol="testsymbol", treasuryAccountId=<OPERATOR_ACCOUNT_ID>, adminKey=<VALID_ED25519_PRIVATE_KEY>, commonTransactionParams.signers=[<VALID_ED25519_PRIVATE_KEY>]                      | The token creation succeeds and the token has the new ED25519 private key as its admin key.        | N                 |
+| 4       | Creates a token with a valid ECDSAsecp256k1 private key as its admin key                                         | name="testname", symbol="testsymbol", treasuryAccountId=<OPERATOR_ACCOUNT_ID>, adminKey=<VALID_ECDSA_SECP256K1_PRIVATE_KEY>, commonTransactionParams.signers=[<VALID_ECDSA_SECP256K1_PRIVATE_KEY>]      | The token creation succeeds and the token has the new ECDSAsecp256k1 private key as its admin key. | N                 |
```
[the remark this answers is not in the document]
Comment (rwalworth) [reply test-specifications/token-service/tokenCreateTransaction.md:295]: Fixed

File: test-specifications/token-service/tokenUpdateTransaction.md (hunk 1 above)
Comment (rwalworth) [reply test-specifications/token-service/tokenUpdateTransaction.md]: Ah good to know, I wasn't aware that's how it worked. I added a test for each "role" key to test this behavior.

File: test-specifications/token-service/tokenCreateTransaction.md [hunk 4]
```diff
@@ -0,0 +1,1158 @@
...
+|---------|-------------------------------------|-----------------------------------------------------------------------------------------------------------------------|------------------------------------------------------------------------------------------|-------------------|
+| 1       | Creates a fungible token            | name="testname", symbol="testsymbol", treasuryAccountId=<OPERATOR_ACCOUNT_ID>, tokenType="ft"                         | The token creation succeeds and the token is a fungible token.                           | N                 |
+| 2       | Creates an NFT                      | name="testname", symbol="testsymbol", treasuryAccountId=<OPERATOR_ACCOUNT_ID>, supplyKey=<VALID_KEY>, tokenType="nft" | The token creation succeeds and the token is an NFT.                                     | N                 |
+| 3       | Creates an NFT without a supply key | name="testname", symbol="testsymbol", treasuryAccountId=<OPERATOR_ACCOUNT_ID>, tokenType="nft"                        | The token creation fails with an TOKEN_HAS_NO_SUPPLY_KEY response code from the network. | N                 |
```
Comment (0xivanov) [inline test-specifications/token-service/tokenCreateTransaction.md:741]: It's this the same as `Creates an NFT with an initial supply of zero without a supply key`

Comment (0xivanov) [reply test-specifications/token-service/tokenCreateTransaction.md:741]: Also, we could add `Creates a fungible token without a supply key` test case

File: test-specifications/token-service/tokenCreateTransaction.md [hunk 5]
```diff
@@ -0,0 +1,1158 @@
...
+  "jsonrpc": "2.0",
+  "id": 12,
+  "result": {
+    "tokenId": "0.0.541",
+    "status": "SUCCESS"
+  }
+}
+```
+
+### **Custom Fees:**
```
Comment (0xivanov) [inline test-specifications/token-service/tokenCreateTransaction.md:900]: Can add test case for response code `CUSTOM_FEE_NOT_FULLY_SPECIFIED` where a custom fee schedule entry did not specify either a fixed or fractional fee

File: test-specifications/token-service/tokenCreateTransaction.md (hunk 4 above)
Comment (rwalworth) [reply test-specifications/token-service/tokenCreateTransaction.md]: > It's this the same as `Creates an NFT with an initial supply of zero without a supply key`

Didn't realize the redundancy here, removed here and in #228 as well.

> Also, we could add `Creates a fungible token without a supply key` test case

I believe most, if not all, fungible tokens created in these tests are created without supply keys. Is there something more specific you would want this to test?

File: test-specifications/token-service/tokenCreateTransaction.md (hunk 5 above)
Comment (rwalworth) [reply test-specifications/token-service/tokenCreateTransaction.md:899]: I don't think it's possible to get this response code using the SDKs... I double checked with services, this response code is given when a `CustomFee` protobuf does not have its fee `oneof` set. Since its impossible in the SDKs to instantiate a `CustomFee` that isn't a `CustomFixedFee`, `CustomFractionalFee`, or `CustomRoyaltyFee`, then I think its impossible to get that response code using the SDKs.

Comment (0xivanov) [reply test-specifications/token-service/tokenCreateTransaction.md:899]: Oh I see.

**Review by 0xivanov — APPROVED** [review]
LGTM

**Review by ivaylogarnev-limechain — APPROVED** [review]
LGTM
